### ISSUE TITLE
SegFault : `print_queue`

### ISSUE THREAD

**Issue by kshitij10496** [OrkoHunter/nxcpy#14]
`print_queue()` gives segmentation fault when printing an empty queue.
I think we are missing a base case.

I will fix it tomorrow. :)

**Comment by OrkoHunter** [comment]
I don't believe #7 as of now fixes the problem. First of all, your `init_attributes()` is a function with return type `struct attributes *` but you return nothing. That's vague.

Secondly, if you "init attributes" from inside of the function, it would not be possible for print_attributes to get the "allocated" pointer `nodes`, hence the SEGFAULT is expected.

**Comment by OrkoHunter** [comment]
So, I suggest not to use `malloc` inside `bfs.c`. Let the allocated array of struct be passed inside the function.

**Comment by OrkoHunter** [comment]
Also, the list of node attributes will be used in the cython module to check for `is_connected`. So, it's better if the list is passed from cython itself.

**Comment by kshitij10496** [comment]
- `init_attributes()` should be a `void` function.
- There is no problem with `print_attributes`. Since I am mallocing `nodes` in `bfs()`, its pointer can be passed to any function call.

**Comment by kshitij10496** [comment]
+1 for passing malloced `attributes` from cython itself.

**Comment by kshitij10496** [comment]
@OrkoHunter I dont think this issue is resolved.
We need to discuss about this

**Comment by OrkoHunter** [comment]
The `print_queue` is working right? Help me understand.

**Comment by kshitij10496** [comment]
Nope.
The `print_queue()` aint working right.

Suppose you want to print an empty queue, it would give a SegFault !

**Comment by kshitij10496** [comment]
We need to raise an error that time or an appropriate print statement

**Comment by OrkoHunter** [comment]
Oops! I confused it with `print_graph` of bfs. Sorry.

**Comment by kshitij10496** [comment]
Cool !
I am working on it
